### ISSUE TITLE
dynamic non selectable row by data of each row?

### ISSUE THREAD

**Issue by mrclaudiotrocelo** [react-bootstrap-table/react-bootstrap-table2#1297]
It's possibile to dynamic say not selectable row by data of row?

**Comment by Ymbere** [comment]
Yes, it is possible to create a helper function to filter the lines with the values ​​that you do not want to select.

You can see an example below and customize it according to your needs.

### Example

```jsx
// Your component stuff...

let notSelectableRows = [];

const getNonSelectableRows = (payload, key, stringToCheck) => {
    let nonSelectable = [];

    payload.forEach((obj, index) => {
        let value = obj[key];
        if (value.includes(stringToCheck)) {
            nonSelectable.push(index);
        }
    });

    return nonSelectable;
};

notSelectableRows = getNonSelectableRows({your payload}, {the key}, {the value to check});

const selectRow = {
    ...
    nonSelectable: notSelectableRows,
    ...
};

<BootstrapTable
    ...
    selectRow={selectRow}
    ...
/>
```

**Comment by lancepng0397** [comment]
@Ymbere may I know how do you get the key for your function? The payload should be whatever I put as my BootstrapTable's data and the value to check should be whatever value I put it as, but how do I get the key?

**Comment by Ymbere** [comment]
@lancepng0397 The key is the key for the value that you want to check on your payload, lets say that your payload is an array of objects like this.
```
id: number,
name: string,
price: number
```
If you want to validate by name your key should be `name`.
You can check this example

[![Edit unruffled-sara-0f2su-1206](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/unruffled-sara-0f2su-1206-0f2su?fontsize=14&hidenavigation=1&theme=dark)

**Comment by lancepng0397** [comment]
@Ymbere Now I finally understand it, thanks a lot! You've saved me a lot of time with your help.
